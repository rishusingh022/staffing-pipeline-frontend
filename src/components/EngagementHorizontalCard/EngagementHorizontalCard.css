.engagement-card-container{
  width: 100%;
  display: flex;
  height: 100px;
  background-color: white;
  min-width: 250px;
  position: relative;
  align-items: center;
}

.engagement-img{
  height: 100%;
  aspect-ratio: 1/1;
}

.engagement-details{
  display: flex;
  flex-direction: column;
  /* gap: 10px; */
  padding: 16px 12px;
  width: 100%;
}

.engagement-id{
  font-size: small;
  font-weight: 400;
}
.engagement-title{
  font-size: 16px;
  font-weight: 600;
}

.engagement-bottom-btn{
  display: flex;
  justify-content: flex-end;
  position: absolute;
  bottom: 0;
  right: 0;
  padding: 10px;
}

.engagement-bottom-btn > div{
  display: flex;
  align-items: center;
  gap: 5px;
  cursor: pointer;
}

.engagement-bottom-btn p {
  font-size: 13px;
  font-weight: 600;
}