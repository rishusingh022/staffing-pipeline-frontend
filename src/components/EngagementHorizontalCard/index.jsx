import React from 'react';
import './EngagementHorizontalCard.css';
import ArrowIcon from '../../assets/images/Vector.png';
import ProjectImage from '../../assets/images/engagement-default.png';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

function EngagementHorizontalCard({ engagementImage, engagementId, engagementTitle }) {
  const navigate = useNavigate();
  const handleViewMore = () => {
    navigate(`/projects/${engagementId}`);
  };
  return (
    <div className="engagement-card-container">
      <img className="engagement-img" src={engagementImage ? engagementImage : ProjectImage}></img>
      <div className="engagement-details">
        <p className="engagement-id">{engagementId}</p>
        <p className="engagement-title">{engagementTitle}</p>
        <div className="engagement-bottom-btn" onClick={handleViewMore}>
          <div>
            <p>View</p>
            <img src={ArrowIcon}></img>
          </div>
        </div>
      </div>
    </div>
  );
}

EngagementHorizontalCard.propTypes = {
  engagementImage: PropTypes.string,
  engagementId: PropTypes.string,
  engagementTitle: PropTypes.string,
};

export default EngagementHorizontalCard;
