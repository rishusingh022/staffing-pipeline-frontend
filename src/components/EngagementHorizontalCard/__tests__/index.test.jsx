import React from 'react';
import EngagementHorizontalCard from '..';
import { render } from '@testing-library/react';

const mockNavigate = jest.fn();
const mockUseParams = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useParams: () => mockUseParams,
}));

describe('EngagementHorizontalCard', () => {
  it('should render', () => {
    const { asFragment } = render(
      <EngagementHorizontalCard
        engagementId={'123456'}
        engagementTitle={'Project'}
        engagementImage={'EngagementImage'}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
