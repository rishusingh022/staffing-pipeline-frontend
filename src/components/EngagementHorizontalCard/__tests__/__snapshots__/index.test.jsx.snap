// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngagementHorizontalCard should render 1`] = `
<DocumentFragment>
  <div
    class="engagement-card-container"
  >
    <img
      class="engagement-img"
      src="EngagementImage"
    />
    <div
      class="engagement-details"
    >
      <p
        class="engagement-id"
      >
        123456
      </p>
      <p
        class="engagement-title"
      >
        Project
      </p>
      <div
        class="engagement-bottom-btn"
      >
        <div>
          <p>
            View
          </p>
          <img
            src="Vector.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
